src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -1335,7 +1351,20 @@ internal virtual HttpResponseMessage GetResponse(HttpClient client, HttpRequestM
                 // When MaximumRetryCount is not specified, the totalRequests is 1.
                 if (totalRequests > 1 && ShouldRetry(response.StatusCode))
                 {
-                    int retryIntervalInSeconds = WebSession.RetryIntervalInSeconds;
+                    double retryIntervalInSeconds = (double)WebSession.RetryIntervalInSeconds;
+                    int exponent = WebSession.MaximumRetryCount - totalRequests + 1;
+
+                    if (String.Equals(RetryMode, "Exponential", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // If RetryMode is specified as "Exponential", use the exponential backoff algorithm to determine the retry interval.
+                        retryIntervalInSeconds = Math.Min(retryIntervalInSeconds * Math.Pow(2.0, exponent), _maximumRetryIntervalInSeconds);
+                    }
+                    else if (String.Equals(RetryMode, "ExponentialJitter", StringComparison.OrdinalIgnoreCase))
+                    {
+                        // If RetryMode is specified as "ExponentialJitter", use the exponential backoff with jitter algorithm to determine the retry interval.
+                        Random random = new Random();
+                        retryIntervalInSeconds = Math.Min(retryIntervalInSeconds * Math.Pow(2.0, exponent) * random.NextDouble(), _maximumRetryIntervalInSeconds);
+                    }
```

Comment:
```suggestion
                    float retryIntervalInSeconds = WebSession.RetryIntervalInSeconds;
                    int exponent = WebSession.MaximumRetryCount - totalRequests + 1;

                    if (string.Equals(RetryMode, "Exponential", StringComparison.OrdinalIgnoreCase))
                    {
                        // If RetryMode is specified as "Exponential", use the exponential backoff algorithm to determine the retry interval.
                        retryIntervalInSeconds = Math.Min(retryIntervalInSeconds * MathF.Pow(2, exponent), _maximumRetryIntervalInSeconds);
                    }
                    else if (string.Equals(RetryMode, "ExponentialJitter", StringComparison.OrdinalIgnoreCase))
                    {
                        // If RetryMode is specified as "ExponentialJitter", use the exponential backoff with jitter algorithm to determine the retry interval.
                        Random random = new();
                        retryIntervalInSeconds = Math.Min(retryIntervalInSeconds * MathF.Pow(2, exponent) * random.NextDouble(), _maximumRetryIntervalInSeconds);
                    }
```

Reply:
The build is failing because I forgot `random.NextDouble()` --> `random.NextSingle()`

Reply:
Another suggestion: Use `Random.Shared.NextSingle()` instead and delete `Random random = new();`

Reply:
> Another suggestion: Use `Random.Shared.NextSingle()` instead and delete `Random random = new();`

Thanks, I didn't know that.